qutebrowser/config/configfiles.py

```diff
@@ -106,6 +110,11 @@ def _save(self):
 
     def load(self):
         """Load self.values from the configured YAML file."""
+        self._initial_values = self._load()
+        self.values = copy.deepcopy(self._initial_values)
```

Comment:
I generally try to avoid copying things around if possible. I think it'd be a much nicer solution to make `self.values` private (`self._values`), add a `get_values()` and `set_value(name, value)` method, and then make that set some kind of `self._dirty` flag. I think adding those methods will be required for #2982 as well anyways, as we need some `changed` signal for that.

Reply:
That's the solution I had in mind, first, but then I realized it'd change the API, so I chose for this lesser API-breaking solution.

Possible solutions, kindly indicate which you would prefer:
- The current one
- Your suggestion
- Your suggestion, using `@property` to avoid API changes
- Saving & comparing a hash of the config dictionary instead of the entire dictionary.

Reply:
I'd prefer mine - I'm always fine with changing API as long as it's not exposed to users. `@property` has some issues with PyQt (it swallows exceptions in some weird way, IIRC), so I tend to avoid it.

So yeah, don't worry about the API, and then pick the simplest one - that'd be 2), IMHO.

Reply:
I implemented method 2. To avoid people modifying `get_values()` without getting the config dirty, I implemented the required dictionary methods directly.

Reply:
That seems a nicer option API-wise as well, I didn't think of that!